### PR TITLE
Native ElasticSearch Driver

### PR DESCRIPTION
**Pull request by chad-codeworkshop**
**Check List**
- [ x] Tests has been run in packages where changes made if available
- [ x] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**
#174 

**Description of Changes Made (if issue reference is not provided)**
Updated Typings and QueryBuilder to include elasticsearchcloud
Implemented ElasticSearchCloudQuery.js for Native ElasticSearch dialect
Updated ElasticSearchDriver to include cloud config for elasticsearch cloud

Bug Fix on TS typings, renamed `member` to `dimension`

### CODE REVIEW

**Comment by ifokeev** [comment]
@paveltiunov no styleguide = everybody style his code as he wants

**Comment by chad-codeworkshop** [comment]
@ifokeev To be honest I was struggling with this as prettier attempts to reformat everything, I eventually just switched it off

**Comment by ifokeev** [comment]
@chad-codeworkshop cubejs have to integrate eslint rules and maintainers need follow this rules too. That's the reason of my message

**Comment by paveltiunov** [comment]
@chad-codeworkshop Hey Chad! Thanks for contributing it! Overall looks good to me! What's elastic search cloud driver? Is it X-Pack SQL or something else?

Also could you please rebase and revert all auto formatting changes? Those really hard to digest and make impossible to use git blame over time. Formatting of blocks you changed is totally ok as soon as it passes linter.

@ifokeev Hey Ivan! Thanks for heads up! We actually have style guide in place: https://github.com/cube-js/cube.js/blob/master/CONTRIBUTING.md#style-guides. Linter isn't part of CI build for all the packages right now however I just activated it for schema-compiler and it's enabled for elastic driver as well.

**Comment by chad-codeworkshop** [comment]
@paveltiunov Hey Pavel, I currently work with the cloud hosted version of ElasticSearch provided by elastic themselves, there seems to be 3 different dialects, AWS-ES, Native ES and, OpenDistro

As you said in slack, the current implementation only works with AWS ES, I implemented the Dialect thats documented by ElasticSearch and the same version they run in their cloud offering

I also was not sure of the naming conventions here as they are going to get confusing in the future especially since the current one is `odelasticsearch` but only works with AWS-ES

I'll sort out the formatting changes today and update the pull request

**Comment by paveltiunov** [comment]
@chad-codeworkshop Hey Chad! Cool! Looks great! I think it makes sense to rename dialect just to ElasticSearch as X-Pack SQL is default and native to elastic unless cloud dialect is different from it. `odelasticsearch` meant to be the single dialect shared between OpenDistro and AWS ES but turns out it isn't the case and seems we'd need to support those separately.

File: packages/cubejs-server-core/core/index.d.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ declare module "@cubejs-backend/server-core" {
     | "sqlite";
 
   export interface QueryFilter {
-    member: string;
+    dimension: string;
```
Comment (paveltiunov) [inline packages/cubejs-server-core/core/index.d.ts:103]: `member` and `dimension` are both allowed however `dimension` is deprecated here.

**Comment by chad-codeworkshop** [comment]
@paveltiunov I have renamed everything to elasticsearch, If I get a bit of time I might look at the OpenDistro dialect, any recommendations for the naming there, since we already have odelasticsearch?

**Comment by paveltiunov** [comment]
@chad-codeworkshop Hey Chad! Looks good to me! I believe we can move `odelasticsearch` to AWS one and create new `odelasticsearch`. It isn't backward compatible but currently driver itself just in preview so it makes sense to do this swap.

**Pull request merged**